tests/integ/scala/test_datatype_suite.py

```diff
@@ -469,29 +458,54 @@ def test_structured_dtypes_iceberg(
         assert df.schema == expected_schema
         assert df.dtypes == expected_dtypes
 
-        # Try to save_as_table
-        structured_type_session.table(table_name).write.save_as_table(
-            save_table_name, iceberg_config=ICEBERG_CONFIG
-        )
-
         save_ddl = structured_type_session._run_query(
-            f"select get_ddl('table', '{save_table_name}')"
+            f"select get_ddl('table', '{table_name}')"
         )
         assert save_ddl[0][0] == (
-            f"create or replace ICEBERG TABLE {save_table_name.upper()} (\n\t"
+            f"create or replace ICEBERG TABLE {table_name.upper()} (\n\t"
             "MAP MAP(STRING, LONG),\n\tOBJ OBJECT(A STRING, B DOUBLE),\n\tARR ARRAY(DOUBLE)\n)\n "
             "EXTERNAL_VOLUME = 'PYTHON_CONNECTOR_ICEBERG_EXVOL'\n CATALOG = 'SNOWFLAKE'\n "
             "BASE_LOCATION = 'python_connector_merge_gate/';"
         )
 
     finally:
         structured_type_session.sql(f"drop table if exists {table_name}")
-        structured_type_session.sql(f"drop table if exists {save_table_name}")
 
 
 @pytest.mark.skipif(
     "config.getoption('local_testing_mode', default=False)",
-    reason="strucutred types do not fully support structured types yet.",
+    reason="local testing does not fully support structured types yet.",
+)
+def test_structured_dtypes_iceberg_create_from_values(
+    structured_type_session, local_testing_mode, structured_type_support
+):
+    if not (
+        structured_type_support
+        and iceberg_supported(structured_type_session, local_testing_mode)
+    ):
+        pytest.skip("Test requires iceberg support and structured type support.")
+
+    _, __, expected_schema = STRUCTURED_TYPES_EXAMPLES[True]
+    table_name = f"snowpark_structured_dtypes_{uuid.uuid4().hex[:5]}"
```

Comment:
just curious, any reason why we are using this naming format instead os standard function `Utils.random_table_name()`?

Reply:
No particular reason. Other tests in this file did it this way so I was mostly just copying style.